Source/ChocolateyGui.Common.Windows/Services/ConfigFileWatcher.cs

```diff
@@ -0,0 +1,53 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="Chocolatey" file="ConfigFileWatcher.cs">
+//   Copyright 2017 - Present Chocolatey Software, LLC
+//   Copyright 2014 - 2017 Rob Reynolds, the maintainers of Chocolatey, and RealDimensions Software, LLC
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+using System.IO;
+using Caliburn.Micro;
+using chocolatey.infrastructure.app.configuration;
+using chocolatey.infrastructure.services;
+using ChocolateyGui.Common.Models.Messages;
+using ChocolateyGui.Common.Services;
+using IFileSystem = chocolatey.infrastructure.filesystem.IFileSystem;
+
+namespace ChocolateyGui.Common.Windows.Services
+{
+    public class ConfigFileWatcher : IConfigFileWatcher
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IEventAggregator _eventAggregator;
+        private readonly FileSystemWatcher _fileSystemWatcher;
+        private readonly IXmlService _xmlService;
+        private readonly string _configFile = chocolatey.infrastructure.app.ApplicationParameters.GlobalConfigFileLocation;
+        private int _lastKnownConfigFileHash;
+
+        public ConfigFileWatcher(IFileSystem fileSystem, IEventAggregator eventAggregator, IXmlService xmlService)
+        {
+            _fileSystem = fileSystem;
+            _eventAggregator = eventAggregator;
+            _xmlService = xmlService;
+            _fileSystemWatcher = new FileSystemWatcher
+            {
+                Path = _fileSystem.get_directory_name(_configFile),
+                Filter = _fileSystem.get_file_name(_configFile),
+                NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.FileName,
+            };
+            _fileSystemWatcher.Changed += ConfigFileChanged;
+            _fileSystemWatcher.EnableRaisingEvents = true;
+            _lastKnownConfigFileHash = _xmlService.deserialize<ConfigFileSettings>(_configFile).GetHashCode();
+        }
+
+        private void ConfigFileChanged(object sender, FileSystemEventArgs e)

```

Comment:
Is there a reason that this is private, and why `ConfigFileChanged` isn't added to the `IConfigFileWatcher` interface?

Reply:
I think because the code I was looking at in Chocolatey Agent had used a private method for the file changed handler that I just kept it private. Given the IConfigFileWatcher, it probably makes sense to make it public and add it to the interface.